compiler/qsc_data_structures/src/namespaces/tests.rs

```diff
@@ -0,0 +1,365 @@
+                        ),
+                    },
+                },
+            ),
+            (
+                [
+                    "ns0",
+                    "nsc",
+                ],
+                RefCell {
+                    value: NamespaceTreeNode {
+                        children: {},
+                        id: NamespaceId(
+                            10,
+                        ),
+                    },
+                },
+            ),
+            (
+                [
+                    "ns1",
+                    "nsa",
+                ],
+                RefCell {
+                    value: NamespaceTreeNode {
+                        children: {},
+                        id: NamespaceId(
+                            12,
+                        ),
+                    },
+                },
+            ),
+            (
+                [
+                    "ns1",
+                    "nsb",
+                ],
+                RefCell {
+                    value: NamespaceTreeNode {
+                        children: {},
+                        id: NamespaceId(
+                            13,
+                        ),
+                    },
+                },
+            ),
+            (
+                [
+                    "ns1",
+                    "nsc",
+                ],
+                RefCell {
+                    value: NamespaceTreeNode {
+                        children: {},
+                        id: NamespaceId(
+                            14,
+                        ),
+                    },
+                },
+            ),
+            (
+                [
+                    "ns2",
+                    "nsa",
+                ],
+                RefCell {
+                    value: NamespaceTreeNode {
+                        children: {},
+                        id: NamespaceId(
+                            16,
+                        ),
+                    },
+                },
+            ),
+            (
+                [
+                    "ns2",
+                    "nsb",
+                ],
+                RefCell {
+                    value: NamespaceTreeNode {
+                        children: {},
+                        id: NamespaceId(
+                            17,
+                        ),
+                    },
+                },
+            ),
+            (
+                [
+                    "ns2",
+                    "nsc",
+                ],
+                RefCell {
+                    value: NamespaceTreeNode {
+                        children: {},
+                        id: NamespaceId(
+                            18,
+                        ),
+                    },
+                },
+            ),
+        ]
+    "#]]
+    .assert_debug_eq(&result_buf);
+}
+// test that after inserting lots of namespaces, all ids are unique and sequential
+#[test]
+fn test_insert_or_find_namespace() {
```

Comment:
This test makes me realize that `insert_or_find_namespace` can create multiple nodes, creating multiple ids, but only returns the id of the leaf node created. But I suppose a feature of this function is that the caller doesn't need to know if the namespace nodes exist or not prior to the call.